lib/parsers/rw2_parser.rb

```diff
@@ -0,0 +1,61 @@
+require_relative 'exif_parser'
+
+class FormatParser::RW2Parser
+  include FormatParser::IOUtils
+  include FormatParser::EXIFParser
+
+  def likely_match?(filename)
+    filename =~ /\.(rw2|raw|rwl)$/i
+  end
+
+  def call(io)
+    @buf = FormatParser::IOConstraint.new(io)
+
+    return unless matches_rw2_definition?
+
+    @buf.seek(0)
+    exif = exif_from_tiff_io(@buf)
+    return unless exif
+
+    # RW2 doesn't use the standard EXIF width and height tags (🤷🏻). We can compute them from the sensor
+    # top/bottom/left/right border tags. See https://exiftool.org/TagNames/PanasonicRaw.html for more.
+    w = sensor_border(exif, :right) - sensor_border(exif, :left)
+    h = sensor_border(exif, :bottom) - sensor_border(exif, :top)
+
+    FormatParser::Image.new(
+      format: :rw2,
+      width_px: w,
+      height_px: h,
+      display_width_px: exif.rotated? ? h : w,
+      display_height_px: exif.rotated? ? w : h,
+      orientation: exif.orientation_sym,
+      intrinsics: { exif: exif },
+      content_type: PANASONIC_RAW_MIMETYPE,
+    )
+  rescue EXIFR::MalformedTIFF
+    nil
+  end
+
+  private
+
+  PANASONIC_RAW_MIMETYPE = 'image/x-panasonic-raw'
```

Comment:
These constants are **not** private. Don't think they need to be, thus I'd move them to the usual spot (under the class declaration). Otherwise, there's an _opinionated_ modifier for that.